apps/website/src/server/trpc/router/clips.ts

```diff
@@ -0,0 +1,142 @@
+import { z } from "zod";
+import { TRPCError } from "@trpc/server";
+import { publicProcedure, router } from "@/server/trpc/trpc";
+
+import {
+  addClip,
+  addVote,
+  clipSubmitInputSchema,
+  getClips,
+  removeVote,
+} from "@/server/db/clips";
+import { getClipDetails } from "@/server/utils/twitch-api";
+
+export type SortOptions = "new" | "top" | "top_7day" | "top_30day";
+
+export const clipsRouter = router({
+  addClip: publicProcedure
+    .input(clipSubmitInputSchema)
+    .mutation(async ({ ctx, input }) => {
+      // https://clips.twitch.tv/slug?query
+      // https://www.twitch.tv/channel/clip/slug?query
+      // https://www.twitch.tv/clip/slug?query
+      // https://m.twitch.tv/channel/clip/slug?query
+
+      const clipRegex =
+        /^https?:.+(?:clips\.twitch\.tv|(?:m\.)?twitch\.tv\/(?:clip|.+\/clip))\/?([\w\d-]+)/;
```

Comment:
```suggestion
        /^https?:.+(?:clips\.twitch\.tv|(?:m\.)?twitch\.tv\/(?:clip|.+\/clip))\/?([\w-]+)/;
```

`\w` already includes `[0-9]`